src/hooks/usePositioning.tsx

```diff
@@ -1,74 +1,99 @@
-import {ElementRef, MutableRefObject, useEffect, useRef, useState} from 'react';
+import {MutableRefObject, useEffect, useRef, useState} from 'react';
 import toPX from 'to-px';
 
-type TPosition = {
+type Position = {
     top?: number;
     left?: number;
     bottom?: number;
     right?: number;
 }
-type TAnchorElOrigin = {
+type AnchorElOrigin = {
     horizontal: 'left' | 'center' | 'right';
     vertical: 'top' | 'center' | 'bottom';
 };
-type TTransformElOrigin = {
+type TransformElOrigin = {
     horizontal: 'left' | 'right';
     vertical: 'top' | 'bottom';
 };
 
-const initialPosition: TPosition = {
+const initialPosition: Position = {
     top: -1000,
     left: -1000
 };
 
-function getPosition(anchorPosition: TPosition): TPosition {
+function getPosition(anchorPosition: Position): Position {
     return {
         top: typeof anchorPosition.top === 'string' ? toPX(anchorPosition.top) : anchorPosition.top,
         left: typeof anchorPosition.left === 'string' ? toPX(anchorPosition.left) : anchorPosition.left
     };
 }
 
+function getClosestRelativeAncestor(el: HTMLElement) {
+    while (el.parentElement) {
+        if (window.getComputedStyle(el).position === 'relative') {
+            return el;
+        }
+        el = el.parentElement;
+    }
+    return el;
+}
+
 function getPositionRelativeToEl(
     resolvedAnchorEl: HTMLDivElement,
-    anchorElOrigin: TAnchorElOrigin,
-    transformElOrigin: TTransformElOrigin,
-    anchorPosition: TPosition
+    anchorElOrigin: AnchorElOrigin,
+    transformElOrigin: TransformElOrigin,
+    anchorPosition: Position,
+    position: string
 ) {
     const anchorElRectangle = resolvedAnchorEl.getBoundingClientRect();
-
-    const point: {top?: number, left?: number} = {};
-
-    // Set vertical origin position
-    if (anchorElOrigin.vertical === 'top') {
-        point.top = anchorElRectangle.top;
-    } else if (anchorElOrigin.vertical === 'center') {
-        point.top = anchorElRectangle.top + (anchorElRectangle.height / 2);
-    } else if (anchorElOrigin.vertical === 'bottom') {
-        point.top = anchorElRectangle.bottom;
+    const point: Position = {};
```

Comment:
If we have the closeRelativeAncestor param as suggested below, we won't need to change anything here and even have the position param - just use a new rect instead of anchorElRectangle , like `new DomRect(resolvedAnchorEl.x - closeRelativeAncestor.x , resolvedAnchorEl.y - closeRelativeAncestor.y, resolvedAnchorEl.width, resolvedAnchorEl.height)`